### PR TITLE
Fix some JPA TCK issues

### PR DESCRIPTION
**Pull request by beikov**
**Fixes Issue**
I didn't create issues for these problems yet, but maybe that's not even necessary. We're in the process of certifying Hibernate ORM 7 against JPA 3.2 and these are some problems of the TCK that we encountered when testing with PostgreSQL.

**Related Issue(s)**
-

**Describe the change**
See the commit messages

**Additional context**
-

CC @alwin-joseph @anajosep @arjantijms @cesarhernandezgt @dblevins @m0mus @edbratt @gurunrao @jansupol @jgallimore @kazumura @kwsutter @LanceAndersen @bhatpmk @RohitKumarJain @shighbar @gthoman @brideck @OndroMih @dmatej
@starksm64 @scottmarlow

### CODE REVIEW

File: jpa/tck-dist/artifact-install.bat [hunk 1]
```diff
@@ -0,0 +1,43 @@
+@echo off
+REM
+REM  Copyright (c) 2008, 2018 Oracle and/or its affiliates. All rights reserved.
```
Comment (scottmarlow) [inline jpa/tck-dist/artifact-install.bat:3]: ` 2008, 2018` should be changed to just `2024` since this is a new file.

**Comment by scottmarlow** [comment]
please review @lukasj @rfelcman

File: jpa/spec-tests/src/main/java/ee/jakarta/tck/persistence/se/schemaGeneration/annotations/discriminatorColumn/Client.java [hunk 2]
```diff
@@ -131,7 +131,10 @@ public void discriminatorColumnTest() throws Exception {
 		// CREATE TABLE SCHEMAGENSIMPLE (ID INTEGER NOT NULL, PRODUCT_TYPE
 		// VARCHAR(31), PRIMARY KEY (ID))
 
-		pass2 = findDataInFile(f2, "DROP TABLE SCHEMAGENSIMPLE");
+		expected.clear();
+		expected.add("DROP TABLE");
+		expected.add("SCHEMAGENSIMPLE");
```
Comment (lukasj) [inline jpa/spec-tests/src/main/java/ee/jakarta/tck/persistence/se/schemaGeneration/annotations/discriminatorColumn/Client.java:136]: this is likely to change the meaning of the check, for example following file becomes valid:

```
DROP TABLE SOME_TABLE
CREATE TABLE SCHEMAGENSIMPLE
```

given actual check in the `findDataInFile` method is `fileContent.toLowerCase().indexOf(searchString.toLowerCase()) >= 0`

is this intented?

Comment (beikov) [reply jpa/spec-tests/src/main/java/ee/jakarta/tck/persistence/se/schemaGeneration/annotations/discriminatorColumn/Client.java:136]: AFAIU, `findDataInFile` checks if one line contains all the elements of the list, so the mentioned file would not become valid. One line must contain all the elements of that list.

Comment (lukasj) [reply jpa/spec-tests/src/main/java/ee/jakarta/tck/persistence/se/schemaGeneration/annotations/discriminatorColumn/Client.java:136]: ah, you're right

**Review by lukasj — APPROVED** [review]

**Review by gurunrao — APPROVED** [review]

**Pull request merged**